plugins/no-override.js

```diff
@@ -18,22 +15,30 @@ module.exports = stylelint.createPlugin(ruleName, (enabled, options = {}) => {
   const primerMeta = requirePrimerFile('dist/meta.json')
   const availableBundles = Object.keys(primerMeta.bundles)
 
-  const immutableSelectors = new Set()
-  const immutableClassSelectors = new Set()
+  const immutableSelectors = new Map()
+  const immutableClassSelectors = new Map()
   for (const bundle of bundles) {
     if (!availableBundles.includes(bundle)) {
       continue
     }
     const stats = requirePrimerFile(`dist/stats/${bundle}.json`)
-    for (const selector of stats.selectors.values) {
-      immutableSelectors.add(selector)
+    const selectors = stats.selectors.values.filter(selector => !!CLASS_PATTERN.test(selector))
+    for (const selector of selectors) {
+      immutableSelectors.set(selector, bundle)
       for (const classSelector of getClassSelectors(selector)) {
-        immutableClassSelectors.add(classSelector)
+        immutableClassSelectors.set(classSelector, bundle)
       }
     }
   }
 
-  // console.warn(`Got ${immutableSelectors.size} immutable selectors from: "${bundles.join('", "')}"`)
+  const messages = stylelint.utils.ruleMessages(ruleName, {
+    rejected: (rule, {selector, bundle}) => {
+      const suffix = bundle ? ` (found in ${bundle})` : ''
+      return selector
+        ? `"${selector}" should not be overridden in "${rule.selector}"${suffix}.`
```

Comment:
The difference between `selector` and `rule.selector` is a little confusing to me.

Reply:
`immutableSelector` would be more clear IMO